Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -16,44 +16,41 @@
 
 #include "Starboard.h"
 #import <StubReturn.h>
+#include <vector>
 
-#include "Foundation/NSOrderedSet.h"
+#import <Foundation/NSOrderedSet.h>
+#import <Foundation/NSException.h>
+
+@implementation NSOrderedSet {
+    std::vector<StrongId<id>> setContainer;
```

Comment:
That's why the combination vector+map works so well; for ordered and indexed operations, the vector comes into play. The map handles uniqueness and object-based lookup.